homeassistant/components/script/__init__.py

```diff
@@ -63,6 +64,13 @@ def _deprecated_legacy_mode(config):
     return config
 
 
+def _not_supported_in_legacy_mode(config):
+    if config.get(CONF_MODE, SCRIPT_MODE_LEGACY) == SCRIPT_MODE_LEGACY:
+        validate_legacy_mode_actions(config[CONF_SEQUENCE])
```

Comment:
you need to return the new value from a validator.

Reply:
oh nevermind, you just don't return here and it's returned 2 lines below

Reply:
Yeah, and this will be going away anyway as I rip out the legacy stuff.